### ISSUE TITLE
Opening session with SessionOptions.ForConnection fails

### ISSUE THREAD

**Issue by VilleHakli** [JasperFx/marten#2136]
If you try to open session by providing connection with `documentStore.OpenSessionAsync(SessionOptions.ForConnection(connection))`, NotSupportedException is thrown ("Invalid combination of SessionOptions").

It seems that the `SessionOptions.Initialize` does not handle case where `OwnsConnection == false` and `OwnsTransactionLifecycle == true`.

Test case to reproduce the bug:
```csharp
using System;
using System.Threading.Tasks;

using Bug2136;

using Marten;
using Marten.Services;
using Marten.Testing.Harness;

using Npgsql;

using Shouldly;

using Weasel.Core;

using Xunit;

namespace DocumentDbTests.Bugs
{
    public class Bug_2136_opening_session_with_existing_connection_should_work: BugIntegrationContext
    {
        [Fact]
        public async Task should_be_able_to_use_provided_connection()
        {
            using var documentStore = SeparateStore(x =>
            {
                x.AutoCreateSchemaObjects = AutoCreate.All;
                x.Schema.For<TestEntity>();
            });

            await documentStore.Advanced.Clean.DeleteAllDocumentsAsync();

            await using (var session = documentStore.OpenSession())
            {
                session.Store(new TestEntity { Name = "Test" });
                await session.SaveChangesAsync();
            }

            await using var connection = GetConnection(ConnectionSource.ConnectionString);
            await using (var session = await documentStore.OpenSessionAsync(SessionOptions.ForConnection(connection)))
            {
                var entities = await session.Query<TestEntity>()
                    .ToListAsync();

                entities.Count.ShouldBe(1);
            }
        }

        private static NpgsqlConnection GetConnection(string connectionString)
        {
            var connection = new NpgsqlConnection(connectionString);
            connection.Open();
            return connection;
        }
    }
}

namespace Bug2136
{
    public class TestEntity
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
    }
}
```

**Comment by VilleHakli** [comment]
@jeremydmiller Was this issue closed accidentally? This issue is related to `SessionOptions.ForConnection` and not `SessionOptions.ForConnectionString` which was fixed in #2135

**Comment by jeremydmiller** [comment]
I added additional tests after yours, so yes.
